lte/cloud/go/services/lte/obsidian/models/validate.go

```diff
@@ -185,6 +190,45 @@ func (m *NetworkEpcConfigsMobility) validateMobility() error {
 	return nil
 }
 
+func (m *NetworkNgcConfigs) ValidateModel(context.Context) error {
+	if err := m.Validate(strfmt.Default); err != nil {
+		return err
+	}
+	if m.SuciProfiles != nil {
+		if err := m.ValidateSuciProfiles(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+//Validation of SuciProfile
+func (m *NetworkNgcConfigs) ValidateSuciProfiles() error {
+	if err := m.Validate(strfmt.Default); err != nil {
+		return err
+	}
+	if len(m.SuciProfiles) == 0 {
+		return errors.New("SuciProfiles should be non-empty")
+	}
+	for i := 0; i < len(m.SuciProfiles); i++ {
+		if m.SuciProfiles[i] != nil {
+			if len(m.SuciProfiles[i].HomeNetworkPrivateKey) == 0 {
+				return errors.New("HomeNetworkPrivateKey should be non-empty")
+			}
+			if len(m.SuciProfiles[i].HomeNetworkPublicKey) == 0 {
+				return errors.New("HomeNetworkPublicKey should be non-empty")
+			}
+			if m.SuciProfiles[i].HomeNetworkPublicKeyIdentifier == nil {
+				return errors.New("HomeNetworkPublicKeyIdentifier should be non-empty")
+			}
+			if m.SuciProfiles[i].ProtectionScheme == "" {
+				return errors.New("ProtectionScheme should be non-empty")
+			}
+		}
+	}
```

Comment:
Same here -- would be preferable if we can enforce this statically in the swagger schema

Reply:
Implemented in the swagger schema